src/Components/Facilities/Facilities.tsx

```diff
@@ -0,0 +1,424 @@
+import axios from 'axios';
+import { useContext, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Typography, Button, Grid, TextField, Pagination, PaginationItem, IconButton, TablePagination, TableFooter } from '@mui/material';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell, { tableCellClasses } from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import Paper from '@mui/material/Paper';
+import { styled } from '@mui/material/styles';
+import Box from '@mui/material/Box';
+import Modal from '@mui/material/Modal';
+import HighlightOffIcon from '@mui/icons-material/HighlightOff';
+import Divider from '@mui/material/Divider';
+import MenuItem from '@mui/material/MenuItem';
+import Menu, { MenuProps } from '@mui/material/Menu';
+import EditIcon from '@mui/icons-material/Edit';
+import { useForm } from 'react-hook-form';
+import deleteImg from "../../assets/Email.png"
+import Avatar from '@mui/material/Avatar';
+import facilityImg from "../../assets/tx-seztx-pool-21834-square.avif"
+import noData from "../../assets/freepik--Character--inject-70.png"
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import DeleteOutlineOutlinedIcon from '@mui/icons-material/DeleteOutlineOutlined';
+import { AuthContext } from '../../Context/AuthContext';
+import { toast } from 'react-toastify';
+
+
+const style = {
+  position: 'absolute' as 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 400,
+  bgcolor: 'background.paper',
+  boxShadow: 24,
+  borderRadius: 2
+};
+const StyledTableCell = styled(TableCell)(({ theme }) => ({
+  [`&.${tableCellClasses.head}`]: {
+
+    color: theme.palette.common.black
+  },
+  [`&.${tableCellClasses.body}`]: {
+    fontSize: 14,
+  },
+}));
+
+const StyledTableRow = styled(TableRow)(({ theme }) => ({
+  '&:nth-of-type(odd)': {
+    backgroundColor: theme.palette.action.hover,
+  },
+  // hide last border
+  '&:last-child td, &:last-child th': {
+    border: 0,
+  },
+}));
+export default function Facilities() {
+
+  const [showState, setShowState] = useState();
```

Comment:
We should set the initial state and not leave it empty, and this applies to other state variables as well.